compiler/rustc_infer/src/infer/equate.rs

```diff
@@ -144,6 +144,13 @@ impl<'tcx> TypeRelation<'tcx> for Equate<'_, '_, 'tcx> {
                 }
             }
 
+            // Inside the generalizer, we want to related projections by substs eagerly.
+            (&ty::Alias(ty::Projection, a_data), &ty::Alias(ty::Projection, b_data))
+                if self.fields.relate_projections_via_substs =>
```

Comment:
I am somewhat unhappy with this and hope that it can be avoided.  I feel like this is fairly related to #105787 so maybe we can talk about these in sync?
```